crates/extensions/tedge_mqtt_ext/src/lib.rs

```diff
@@ -223,3 +230,59 @@ impl Actor for MqttActor {
         .await
     }
 }
+
+#[async_trait]
+pub trait MqttConnector: Send {
```

Comment:
These concepts of `MqttConnector` and `MqttConnectionFactory` have to be moved to `tedge_agent`.

This will make the dependency graph clearer.